MBS-10299: Add page with suggested edit searches

reosarevok opened the pull request:
https://tickets.metabrainz.org/browse/MBS-10299

This currently includes three sets of links: one for destructive edits, one for unreviewed / controversial edits, and one for edits by beginners.

Once [MBS-7912](https://tickets.metabrainz.org/browse/MBS-7912) is available, it should be possible to also add a filter for these (when relevant, at least) to let the user see edits related to their suscriptions only.

mwiencek commented on root/report/VotingReportsIndex.js:
```diff
@@ -0,0 +1,363 @@
+      </p>
+      <ul>
+        <VotingReportRow
+          noOwnEditsUrl="/search/edits?auto_edit_filter=&order=asc&negation=0&combinator=and&conditions.0.field=type&conditions.0.operator=%3D&conditions.0.args=9&conditions.0.args=84&conditions.0.args=4&conditions.0.args=153&conditions.0.args=134&conditions.0.args=14&conditions.0.args=64&conditions.0.args=74&conditions.0.args=24&conditions.0.args=225%2C223%2C311&conditions.0.args=143&conditions. ...
```
I'd mainly just like to see something done about these URLs if possible. I guess splitting them all across multiple lines would also be horrible and make the file 10,000 lines long, but...maybe there's an alternative. For example, all the `noOwnEditsUrl` URLs probably have a common set of conditions and other parameters to filter out own edits. Perhaps you could extract the common parts of these different URLs into constants, and concatenate them where needed? (So any common conditions included in a constant would have to start with `conditions.0` and end at `conditions.n` for some `n`, and whatever strings use the constant would have to have their other conditions all start at `n + 1`.) This also sounds really tedious though and perhaps not a good use of dev time. So if you don't think any of that is worth the effort I guess we can include it as-is.

reosarevok replied:
I hate the URLs here as much as you do, but tbh I think this should be fixed by fixing our stupid edit search URLs. If you have any idea how to make them less absurdly long, I'd be interested :p Your suggestion would make a lot of sense to me if it wasn't because of the condition numbering, but once it starts requiring modifying the constant to allow for changing condition numberings, it stops sounding like it saves effort / helps readability...

mwiencek replied:
Yeah, perhaps we should add some shortcut parameters for common conditions, like "me_no_vote=1" (expect a better name and example, lol). But something we can improve on later.

yvanzo replied:
Maybe it would be more readable to split by conditions at least, for this example:
```suggestion
          noOwnEditsUrl="/search/edits?" +
          "conditions.0.field=type&conditions.0.operator=%3D&conditions.0.args=9&conditions.0.args=84&conditions.0.args=4&conditions.0.args=153&conditions.0.args=134&conditions.0.args=14&conditions.0.args=64&conditions.0.args=74&conditions.0.args=24&conditions.0.args=225%2C223%2C311&conditions.0.args=143&conditions.0.args=44&conditions.0.args=83&conditions.0.args=3&conditions.0.args=315&conditions.0.args=152&conditions.0.args=133&conditions.0.args=78&conditions.0.args=410&conditions.0.args=13&conditions.0.args=53&conditions.0.args=63&conditions.0.args=73&conditions.0.args=23&conditions.0.args=36&conditions.0.args=224&conditions.0.args=142&conditions.0.args=211&conditions.0.args=43&conditions.0.args=47&" +
          "conditions.1.field=status&conditions.1.operator=%3D&conditions.1.args=1&" +
          "conditions.2.field=editor&conditions.2.operator=not_me"
```

I also removed unneeded defaults.

mwiencek commented on root/report/VotingReportsIndex.js:
```diff
@@ -0,0 +1,363 @@
+                  {l('…that I haven’t voted on yet')}
+                </a>
+              </li>
+            </ul>
```
This `ul` should be merged with the one below, since they're at the same level and part of the same semantic list. It wouldn't look visually different, but def. creates a confusing page structure for screen readers in particular.

There appear to be similar cases below.

mwiencek commented:
Maybe we should default to having all links exclude edits you've voted on, to reduce the number of URL permutations? I expect that'd be the most common use case, and it's quite simple to remove that condition if you want to see ones you already voted on (we could even mention that on the page).

reosarevok commented:
That seems sensible. Made the changes, which make for a much less scary-looking page, and added a note about it at the end of the introduction.

mwiencek commented:
This looks great. :) One small thing I noticed in the text is that we use the phrase "No votes" a lot, but this might be confused to mean either "has votes against" ("No" votes) or "has zero votes" (no votes). Maybe we can keep the "No" in quotes when referencing it as a type of vote?

reosarevok commented:
Did that (for “Yes”, “No” and “Abstain”, since it'd look super weird otherwise).

yvanzo commented on root/layout/components/BottomMenu.js:
```diff
@@ -251,6 +251,9 @@ const EditingMenu = () => (
         <a href="/reports">{l('Reports')}</a>
       </li>
+      <li>
+        <a href="/voting-reports">{l('Voting Reports')}</a>
```
Maybe it would be simpler to just replace the above _Vote on Edits_? Both have the same purpose. The new page is just more sophisticated.

reosarevok replied:
I guess it might make sense - at the very least we should rename "Vote on Edits" to "All Open Edits" or something, because right now it's actually kind of weird. But I'd be fine with replacing the link too - we might just have to add an "All open edits" link to this page.

yvanzo commented on root/layout/components/BottomMenu.js:
```diff
@@ -251,6 +251,9 @@ const EditingMenu = () => (
         <a href="/reports">{l('Reports')}</a>
       </li>
+      <li>
+        <a href="/voting-reports">{l('Voting Reports')}</a>
```
Also, _Voting Reports_ might be misleading as these are searches rather than daily built reports like (data) _Reports_.

reosarevok commented:
I changed this to not use the word "Report". I also replaced the open edits link in the menu, and added both "all open edits" and "all edits" to the bottom of the page (I used an edit search for open edits rather than edit/open because that's easier to refine later).

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,465 @@
+type VotingGuideRowProps = {
+  +guideName: string,
+  +mainUrl: string,
+  +subscribedArtistsUrl?: string,
```
That might be replaced with a flag, as the URL for subscribed artists can be generated by appending `&conditions.9.field=artist&conditions.9.operator=subscribed` to the main URL.

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+
+      <p>
+        {exp.l(
+          `If you'd like to help ensuring the changes made to MusicBrainz data
```
```suggestion
          `If you’d like to help ensuring the changes made to MusicBrainz data
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           impossible to revert. As such, an incorrect destructive edit that
+           applies unnoticed can cause quite a big mess! Most are guaranteed
+           to remain open for at least two full days even if they get three
+           “Yes” votes, to avoid them closing too quickly, but it's always
```
```suggestion
           “Yes” votes, to avoid them closing too quickly, but it’s always
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+          `If you'd like to help ensuring the changes made to MusicBrainz data
+           are correct, but do not know where to start, the following reports
+           should be useful. These are pre-defined
+           {edit_search|edit searches}; once you're comfortable with the edit
```
```suggestion
           {edit_search|edit searches}; once you’re comfortable with the edit
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           {coc|Code of Conduct} in mind. Almost all editors want to help, so
+           your goal as a voter is to help them help better. This applies even
+           more for beginners: always try to be helpful and patient with them,
+           even if they are making mistakes, so that they'll hopefully grow
```
```suggestion
           even if they are making mistakes, so that they’ll hopefully grow
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           can do better in the future. Make sure to link to the appropriate
+           guidelines, or to relevant examples of well-entered data, and above
+           all remember to be nice. Chances are if these users are making a
+           terrible mess, they're not doing it on purpose but out of
```
```suggestion
           terrible mess, they’re not doing it on purpose but out of
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           all remember to be nice. Chances are if these users are making a
+           terrible mess, they're not doing it on purpose but out of
+           confusion: MusicBrainz can be daunting for newcomers! If at all
+           possible, fix the errors (and let the editor know that you've done
```
```suggestion
           possible, fix the errors (and let the editor know that you’ve done
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           that and that they can check the edits you made to see how it
+           should look like) rather than voting “No” on edits, since “No”
+           votes can be quite discouraging, especially as a new editor.
+           If something is just so bad that there's no fixing it and it is
```
```suggestion
           If something is just so bad that there’s no fixing it and it is
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           If something is just so bad that there's no fixing it and it is
+           making the existing data worse, do vote against the edit, but
+           make sure to explain nicely why that is needed, rather than just
+           silently “No”-voting. For a longer overview of the attitude we're
```
```suggestion
           silently “No”-voting. For a longer overview of the attitude we’re
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           silently “No”-voting. For a longer overview of the attitude we're
+           hoping for, see {voting_blog|this blog post about voting}.
+           Also, if a beginner is making especially good edits,
+           you might want to let them know so they'll feel good about it!`,
```
```suggestion
           you might want to let them know so they’ll feel good about it!`,
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+      </p>
+      <p>
+        {l(
+          `The filters for edits you haven't voted on yet below will only show
```
```suggestion
          `The filters for edits you haven’t voted on yet below will only show
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+      <p>
+        {l(
+          `The filters for edits you haven't voted on yet below will only show
+           open edits (since you can't vote on closed edits anyway).
```
```suggestion
           open edits (since you can’t vote on closed edits anyway).
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+          `Sometimes you might want to just check all edits for some reason.
+           If you feel like being overwhelmed by a very long list of edits
+           is just what the doctor ordered, just check the searches below!
+           Don't forget you can always experiment with different edit search
```
```suggestion
           Don’t forget you can always experiment with different edit search
```

mwiencek commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           should be useful. These are pre-defined
+           {edit_search|edit searches}; once you’re comfortable with the edit
+           search you can refine them further or just create your own personal
+           reports for your own use!`,
```
Are we sticking with using the "reports" terminology here?

reosarevok replied:
Whoops. Nope! Fixed.

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+      <p>
+        {l(
+          `Edits that nobody have seen can always benefit from a quick check:
+           even if you're not familiar with the music in question, you might
```
```suggestion
           even if you’re not familiar with the music in question, you might
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           even if you're not familiar with the music in question, you might
+           be able to notice that something seems wrong. Don't forget you
+           don't need to vote on every edit: it's perfectly fine to just
+           abstain if you feel something doesn't seem wrong but it's also not
```
```suggestion
           abstain if you feel something doesn’t seem wrong but it’s also not
```

yvanzo commented on root/vote/VotingIndex.js:
```diff
@@ -0,0 +1,381 @@
+           be able to notice that something seems wrong. Don't forget you
+           don't need to vote on every edit: it's perfectly fine to just
+           abstain if you feel something doesn't seem wrong but it's also not
+           100% obvious that it is right without further checks.`,
```
definitely not “right without further checks” :wink:
